### PR TITLE
Test for chol!(Number)

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: test/linalg/cholesky.jl [hunk 1]
```diff
@@ -89,6 +89,9 @@ end
         apos = apd[1,1]            # test chol(x::Number), needs x>0
         @test all(x -> x ≈ √apos, cholfact(apos).factors)
         @test_throws PosDefException chol(-one(eltya))
+        # test chol! with a Number
+        chol_num = Base.LinAlg.chol!(apos, :U)
+        @test all(x -> x ≈ √apos, chol_num)
```
Comment (Sacha0) [inline test/linalg/cholesky.jl]: Perhaps simply `@test chol!(4, :U) ≈ 2`? :)

**Comment by fredrikekre** [comment]
Do we actually need `chol!(x::Number)`? it is a bit misleading with the `!`. I know we implement `_chol!(x::Number, uplo)` but the purpose of that is for the generic cholfact implementation to work correctly. I don't see chol!(x::Number)` being used anywhere internally, and for users, `chol(x)` should do the exact same thing?

**Comment by mschauer** [comment]
From 
https://github.com/JuliaLang/julia/blob/d9c04ad5288cbbe0f8157995e4a5deae16af026c/test/linalg/symmetric.jl#L166 it appears that `chol!` it is/was used on numbers. In general, for chainable modifying array functions it is useful to treat immutables gracefully, e.g. in `trace(chol!(B*B',:L))` is the point is to allow modification for efficiency, not to change `B*B'`.

**Comment by andreasnoack** [comment]
@mschauer is right. I wanted to write a recursive Cholesky that didn't allocate. See https://github.com/JuliaLang/julia/blob/3b98d233291bfe7549fdbcb3225df104c52d0a81/base/linalg/cholesky.jl#L108-L112. For this to work, I had to define function that updated the in-place when possible. However, now that this is `_chol!`, there is probably no reason to have `chol!`.

**Comment by andreasnoack** [comment]
Why would they need to add methods to `_chol!`?

**Comment by mschauer** [comment]
Because that is the function called on the elements of an array when a cholesky decomposition is performed.

**Comment by andreasnoack** [comment]
Why are the existing methods not sufficient?

**Comment by kshyatt** [comment]
Did we come to a conclusion about whether to keep this or not?

**Comment by fredrikekre** [comment]
I think so? `@deprecate chol!(x::Number, uplo) chol(x)`

**Comment by andreasnoack** [comment]
Superseded by #24498

**Pull request closed without merging**